### PR TITLE
EPAS: DB-2311

### PR DESCRIPTION
**Pull request by ccestes**
## What Changed?

Adding EDB SPL Check extension documentation as per [DB-2311](https://enterprisedb.atlassian.net/browse/DB-2311)

[DB-2311]: https://enterprisedb.atlassian.net/browse/DB-2311?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: advocacy_docs/pg_extensions/spl_check/using/tracer.mdx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+To enable tracing, set `spl_check.tracer` to `on`. 
+
+!!!Warning 
+   Enabling tracing can have a significant impact on performance. To prevent performance issues, you can set a level for outputs used by the tracer with `spl_check.tracer_errlevel`. (The default is `notice`.) You can limit the output content according to the length specified with the `spl_check.tracer_variable_max_length` configuration variable. 
+
+As a security safeguard, we recommend that the use of the tracer be enabled by a superuser by setting `spl_check.enable_tracer` to `on` or `spl_check.enable_tracer` to `on` in `postgresql.conf`. Because the tracer shows the content of EDB SPL Check's variables, some sensitive security information can display for an unprivileged user. 
+
+We also recommend that you load the extension `spl_check`. You can execute some `spl_check` functions explicitly with `load 'spl_check';`. You can use the configuration's `shared_preload_libraries`, `local_preload_libraries`, or `session_preload_libraries` option. 
+
```
Comment (ebgitelman) [inline advocacy_docs/pg_extensions/spl_check/using/tracer.mdx:32]: Check edit (deleted "or"). Last sentence doesn't feel connected to the rest of the paragraph. What can you use those options for?

File: advocacy_docs/pg_extensions/spl_check/index.mdx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+EDB SPL Check is supported on EDB Postgres Advanced Server versions 12 and later.
+
+These are some of the key features of EDB SPL Check:
+
+- Checks fields of referenced database objects and types inside embedded SQL
+- Validates you're using the correct types for function parameters
+- Identifies unused variables and function arguments as well as unmodified `OUT` arguments
+- Partially detects dead code, that is, code after an unqualified `RETURN` command
+- Detects missing `RETURN` commands in functions, which are common after exception handlers or complex logic 
+- Tries to identify unwanted hidden casts, which can be a performance issues like unused indexes
```
Comment (ebgitelman) [inline advocacy_docs/pg_extensions/spl_check/index.mdx:18]: Which can **cause** performance issues?

Comment (ccestes) [reply advocacy_docs/pg_extensions/spl_check/index.mdx:18]: Changing to, "which can cause performance issues like unused indexes."

File: advocacy_docs/pg_extensions/spl_check/using/tracer.mdx (hunk 1 above)
Comment (ccestes) [reply advocacy_docs/pg_extensions/spl_check/using/tracer.mdx:32]: I agree; this was unclear. Changing to: "You can also set the configuration options in `shared_preload_libraries`, `local_preload_libraries`, or `session_preload_libraries` options to load `spl_check`. "

File: advocacy_docs/pg_extensions/spl_check/using/profiler.mdx [hunk 3]
```diff
@@ -0,0 +1,122 @@
+---
+title: Profiler
+---
+
+EDB SPL Check contains a simple profiler of `edbspl` functions and procedures. It can work with or without access to shared memory and depends on the `shared_preload_libraries` config. 
+
+When EDB SPL Check is initialized by `shared_preload_libraries`, it can allocate shared memory and stores function profiles. When EDB SPL Check can't allocate shared memory, the profile is stored in session memory. 
+
+Due to dependencies, `shared_preload_libraries` must contain `edbspl` first: 
+
```
[the remark this answers is not in the document]
Comment (ccestes) [reply advocacy_docs/pg_extensions/spl_check/using/profiler.mdx]: `edbspl` changed to `edb-spl`, which matches the code example.

**Comment by nidhibhammar** [comment]
@ccestes, we need consistent code blocks with capitalised keywords across the doc. I have edited a few code blocks and formatted the output.

File: advocacy_docs/pg_extensions/spl_check/configuring.mdx [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+spl_check.mode = [ disabled | by_function | fresh_start | every_start ]
+spl_check.fatal_errors = [ yes | no ]
+
+spl_check.show_nonperformance_warnings = false
+spl_check.show_performance_warnings = false
+```
+
+By default, `spl_check.mode` is set to `by_function`, which means that checks are done only in active mode by using `spl_check_function`. `fresh_start` means cold start, so function's called first. 
+
+<!-- Query last sentence -->
```
[the remark this answers is not in the document]
Comment (ccestes) [reply advocacy_docs/pg_extensions/spl_check/configuring.mdx]: Yep, it should be that. I switched it.

File: advocacy_docs/pg_extensions/spl_check/index.mdx [hunk 5]
```diff
@@ -0,0 +1,22 @@
+---
+title: EDB SPL Check
+---
+
+EDB SPL Check is a code analysis tool for SPL on EDB Postgres Advanced Server. Using only the internal PostgreSQL parser, it identifies errors that can occur at runtime. In addition, it parses the SQL inside your routines and identifies errors that are usually missed when executing `CREATE PROCEDURE/FUNCTION/PACKAGE`. Similarly, it identifies errors that are usually missed when executing `CREATE PACKAGE/OBJECT TYPES/COMPOUND TRIGGER`.
+
+You can control the levels of many warnings and hints. For instance, you can add `PRAGAMA` type markers to turn certain aspects off or on, allowing you to hide messages you're already aware of or keep ones you want to be reminded of. 
+
+EDB SPL Check is supported on EDB Postgres Advanced Server versions 12 and later.
+<!-- Check this -->
```
[the remark this answers is not in the document]
Comment (ccestes) [reply advocacy_docs/pg_extensions/spl_check/index.mdx]: Which edit is this?

File: advocacy_docs/pg_extensions/spl_check/using/index.mdx [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+         pg_trigger.tgfoid IS NOT NULL)
+    OFFSET 0
+) ss
+ORDER BY (pcf).functionid::regprocedure::text, (pcf).lineno;
+```
+
+## Limitations
+
+EDB SPL Check finds almost all errors on static code. However, when using PL/pgSQL's dynamic features like dynamic SQL or the record-data data type, false positives are possible. In these cases, we recommend that you rewrite the affected function or disable EDB SPL Check for the function. For example: 
+<!-- record data data type ? -->
```
[the remark this answers is not in the document]
Comment (ccestes) [reply advocacy_docs/pg_extensions/spl_check/using/index.mdx]: Good catch. Should just be `record data type`.

File: advocacy_docs/pg_extensions/spl_check/using/pragma.mdx [hunk 7]
```diff
@@ -0,0 +1,87 @@
+---
+title: Pragma
+---
+
+You can configure EDB SPL Check's behavior inside a checked function with a pragma function. This is an analogy of PL/SQL or ADA language of a pragma feature. PL/pgSQL doesn't support pragma, but EDB SPL Check detects functions named `spl_check_pragma` and takes options from the parameters of the function. These EDB SPL Check options are valid to add to the end of this group of statements: 
+
```
[the remark this answers is not in the document]
Comment (ccestes) [reply advocacy_docs/pg_extensions/spl_check/using/pragma.mdx:6]: Was also confused by this sentence. I'm going to ask for clarification from a dev. 

Which edit?

File: advocacy_docs/pg_extensions/spl_check/index.mdx (hunk 5 above)
[the remark this answers is not in the document]
Comment (ebgitelman) [reply advocacy_docs/pg_extensions/spl_check/index.mdx]: Oh, the original isn't showing. It originally said versions 12 to 16.

**Review by nidhibhammar — APPROVED** [review]

**Pull request merged**